unstructured/ingest/v2/processes/connectors/elasticsearch.py

```diff
@@ -158,73 +171,76 @@ def get_identifier(self, index_name: str, record_id: str) -> str:
+        )
+        download_path.parent.mkdir(parents=True, exist_ok=True)
+        try:
+            with open(download_path, "w", encoding="utf8") as f:
+                f.write(self.map_es_results(es_results=result))
+        except Exception as e:
+            logger.error(
+                f"failed to download from index {index_name} "
+                f"and id {record_id} to {download_path}: {e}",
+                exc_info=True,
+            )
+            raise SourceConnectionNetworkError(f"failed to download file {file_data.identifier}")
+        return DownloadResponse(
+            file_data=FileData(
+                identifier=filename_id,
+                connector_type=CONNECTOR_TYPE,
+                metadata=DataSourceMetadata(
+                    version=str(result["_version"]) if "_version" in result else None,
+                    date_processed=str(time()),
+                    record_locator={
+                        "hosts": self.connection_config.hosts,
+                        "index_name": index_name,
+                        "document_id": record_id,
+                    },
+                ),
+            ),
+            path=download_path,
+        )
+
     def run(self, file_data: FileData, **kwargs: Any) -> download_responses:
+        raise NotImplementedError()
+
+    async def run_async(self, file_data: FileData, **kwargs: Any) -> download_responses:
+        from elasticsearch.helpers import async_scan
+
         index_name: str = file_data.additional_metadata["index_name"]
         ids: list[str] = file_data.additional_metadata["ids"]
 
-        es_results = self.get_results(ids=ids, index_name=index_name)
+        scan_query = {
+            "_source": self.download_config.fields,
+            "version": True,
+            "query": {"ids": {"values": ids}},
+        }
+
         download_responses = []
-        for result in es_results:
-            record_id = result["_id"]
-            filename_id = self.get_identifier(index_name=index_name, record_id=record_id)
-            filename = f"{filename_id}.txt"
-            download_path = self.download_config.download_dir / Path(filename)
-            logger.debug(
-                f"Downloading results from index {index_name} and id {record_id} to {download_path}"
-            )
-            download_path.parent.mkdir(parents=True, exist_ok=True)
-            try:
-                with open(download_path, "w", encoding="utf8") as f:
-                    f.write(self.map_es_results(es_results=result))
-            except Exception as e:
-                logger.error(
-                    f"failed to download from index {index_name} "
-                    f"and id {record_id} to {download_path}: {e}",
-                    exc_info=True,
-                )
-                raise SourceConnectionNetworkError(
-                    f"failed to download file {file_data.identifier}"
-                )
+        async for result in async_scan(
```

Comment:
We now get 
```
Unclosed connector
connections: ['[(<aiohttp.client_proto.ResponseHandler object at 0x13658ee60>, 271841.176414041)
```
in the logging. Not sure if we can address this. You would think it would close out the connector when it is done automatically?

Reply:
I'll look into this

Reply:
resolved in latest commits